diff --git i/lua/nlua/ftplugin.lua w/lua/nlua/ftplugin.lua
index 033fb53..1f25a3f 100644
--- i/lua/nlua/ftplugin.lua
+++ w/lua/nlua/ftplugin.lua
@@ -46,5 +46,5 @@ return function()
   --      mapper('n', 'K', '<cmd>lua vim.lsp.buf.hover()<CR>')
   --    end
   -- TODO: Customize keymap
-  vim.api.nvim_buf_set_keymap(0, 'n', 'K', '<cmd>lua nlua.keyword_program()<CR>', {noremap = true, silent = true})
+  vim.api.nvim_buf_set_keymap(0, 'n', 'M', '<cmd>lua nlua.keyword_program()<CR>', {noremap = true, silent = true})
 end
