diff --git i/plugin/UnconditionalPaste.vim w/plugin/UnconditionalPaste.vim
index d39c274..cb7f2b1 100644
--- i/plugin/UnconditionalPaste.vim
+++ w/plugin/UnconditionalPaste.vim
@@ -143,20 +143,26 @@ endif
 "- mappings --------------------------------------------------------------------
 
 function! s:CreateMappings()
+    " \       ['Jagged', 'B'],
+    " \       ['Delimited', 'qb'], ['RecallDelimited', 'QB'],
+    " \       ['Shifted', '>'],
+    " \       ['Commented', '#'],
+    " \       ['MoreIndent', 'm'], ['LessIndent', 'n'],
+    " \       ['Unjoin', 'u'], ['RecallUnjoin', 'U'],
+    " \       ['Plus', 'p'], ['PlusRepeat', '.p'],
+    " \       ['GPlus', 'P'], ['GPlusRepeat', '.P']
+    " \       ['Comma', ','],
+    " \       ['CommaSingleQuote', ",'"],
+    " \       ['CommaDoubleQuote', ',"'],
+    " \       ['Queried', 'q'],
+    " \       ['RecallQueried', 'Q'],
     for [l:pasteName, l:pasteType] in
     \   [
-    \       ['Char', 'c'], ['Line', 'l'], ['Block', 'b'], ['Comma', ','], ['CommaSingleQuote', ",'"], ['CommaDoubleQuote', ',"'],
+    \       ['Char', 'c'],
+    \       ['Line', 'l'],
+    \       ['Block', 'b'],
     \       ['Indented', 'l'],
-    \       ['MoreIndent', 'm'], ['LessIndent', 'n'],
-    \       ['Shifted', '>'],
-    \       ['Commented', '#'],
     \       ['Spaced', 's'],
-    \       ['Jagged', 'B'],
-    \       ['Delimited', 'qb'], ['RecallDelimited', 'QB'],
-    \       ['Queried', 'q'], ['RecallQueried', 'Q'],
-    \       ['Unjoin', 'u'], ['RecallUnjoin', 'U'],
-    \       ['Plus', 'p'], ['PlusRepeat', '.p'],
-    \       ['GPlus', 'P'], ['GPlusRepeat', '.P']
     \   ]
 	for [l:direction, l:pasteCmd] in [['After', 'p'], ['Before', 'P']]
 	    let l:mappingName = 'UnconditionalPaste' . l:pasteName . l:direction
@@ -175,6 +181,8 @@ function! s:CreateMappings()
 			nmap g]P <Plug>UnconditionalPasteIndentedBefore
 			nmap g[P <Plug>UnconditionalPasteIndentedBefore
 			nmap g[p <Plug>UnconditionalPasteIndentedBefore
+			nmap ghp <Plug>UnconditionalPasteIndentedAfter
+			nmap ghP <Plug>UnconditionalPasteIndentedBefore
 		    endif
 		elseif l:pasteName ==# 'MoreIndent'
 		    let l:pasteMappingDefaultKeys = ']]' . l:pasteCmd
@@ -223,11 +231,11 @@ function! s:CreateMappings()
 	endfor
     endfor
 
+    " \       ['Unjoin', 'u', '<C-u>'], ['RecallUnjoin', 'U', '<C-u><C-u>']
     for [l:pasteName, l:pasteType, l:pasteKey] in
     \   [
     \       ['Char', 'c', '<C-c>'], ['Comma', ',', ','],
     \       ['Queried', 'q', '<C-q>'], ['RecallQueried', 'Q', '<C-q><C-q>'],
-    \       ['Unjoin', 'u', '<C-u>'], ['RecallUnjoin', 'U', '<C-u><C-u>']
     \   ]
 	let l:plugMappingName = '<Plug>UnconditionalPaste' . l:pasteName
 	" XXX: Can only use i_CTRL-R here (though I want literal insertion, not
