diff --git a/recentdirs.sh b/recentdirs.sh
index b387f02..7e41634 100755
--- a/recentdirs.sh
+++ b/recentdirs.sh
@@ -16,12 +16,12 @@ if [[ $FZFZ_RECENT_DIRS_TOOL == "z" ]]; then
     source "$SCRIPT_PATH/z.sh"
     _z -l 2>&1 && exit 0 || exit 0
 elif [[ $FZFZ_RECENT_DIRS_TOOL == "autojump" ]]; then
-    if [[ $OSTYPE == darwin* && -z $(whence tac) ]]; then
-        REVERSER='tail -r'
+    if [[ $OSTYPE == darwin* && -n $(type tac) ]]; then
+        REVERSER='/usr/bin/tail -r'
     else
         REVERSER='tac'
     fi
-    autojump -s | $REVERSER | tail +8 | $REVERSER | awk '{print $2}'
+    autojump -s | $REVERSER | /usr/bin/tail +8 | $REVERSER | awk '{print $2}'
 elif [[ $FZFZ_RECENT_DIRS_TOOL == "fasd" ]]; then
     fasd -dl 2>&1 && exit 0 || exit 0
 else
