diff --git i/zsh-edit.plugin.zsh w/zsh-edit.plugin.zsh
index 4817374..664430e 100644
--- i/zsh-edit.plugin.zsh
+++ w/zsh-edit.plugin.zsh
@@ -10,7 +10,12 @@ zsh-edit() {
   local dir=${${(%):-%x}:P:h}
   local fdir=$dir/functions
   typeset -gU FPATH fpath=( $dir $fpath )
-  autoload -Uz $fdir/bind $fdir/[._]edit.*~*.zwc(DN)
+  # autoload -Uz $fdir/bind
+  # autoload -Uz $fdir/_*~*.zwc(:t)
+
+  autoload -Uz $fdir/bind
+  autoload -Uz $fdir/.edit.*~*.zwc(:t)
+  autoload -Uz $fdir/_edit.*~*.zwc(:t)

   .beginning-of-buffer() {
     CURSOR=0
@@ -38,36 +43,40 @@ zsh-edit() {
   .edit.bind() {
     local widget=$1 emacs=$2
     shift 2
-    bindkey -M emacs "$emacs" "$widget"
+    bindkey -M vicmd "$emacs" "$widget"
     bindkey "${@:^^widget}"
   }

-  .edit.bind backward-subword         '^[b'   '^[^['{\[,O}D   '^['{'[1;',\[,O}{3,5}D
-  .edit.bind backward-shell-word      '^[^B'  '^[^['{\[,O}A   '^['{'[1;',\[,O}{3,5}A
-  .edit.bind forward-subword          '^[f'   '^[^['{\[,O}C   '^['{'[1;',\[,O}{3,5}C
-  .edit.bind forward-shell-word       '^[^F'  '^[^['{\[,O}B   '^['{'[1;',\[,O}{3,5}B
-  .edit.bind backward-kill-subword    '^[^?'  '^[^?'  '^H'  '^[[27;'{3,5}';8~'
-  .edit.bind backward-kill-shell-word '^W'    '^[^H'        '^[[27;'{6,7}';8~'
-  bindkey '^[[3~' delete-char
-  .edit.bind kill-subword             '^[d'   '^[(' '^[^[[3~'   '^[[3;'{3,5}\~
-  .edit.bind kill-shell-word          '^[^D'        '^[^[[3;5~' '^[[3;'{6,7}\~
-  .edit.bind beginning-of-line        '^A'    '^['{\[,O}H
-  .edit.bind end-of-line              '^E'    '^['{\[,O}F
-  .edit.bind beginning-of-buffer      '^[<'   '^['{'[1;',\[,O}5H  '^[[5~'
-  .edit.bind end-of-buffer            '^[>'   '^['{'[1;',\[,O}5F  '^[[6~'
-  .edit.bind backward-kill-line       '^U'    '^['{'[1;',\[,O}2H  '^['{'[1;',\[,O}2D
-  .edit.bind kill-line                '^K'    '^['{'[1;',\[,O}2F  '^['{'[1;',\[,O}2C
+  .edit.bind backward-subword         'b'   '^[^['{\[,O}D   '^['{'[1;',\[,O}{3,5}D
+  .edit.bind backward-shell-word      'W'  '^[^['{\[,O}A   '^['{'[1;',\[,O}{3,5}A
+  .edit.bind forward-subword          'e'   '^[^['{\[,O}C   '^['{'[1;',\[,O}{3,5}C
+  .edit.bind forward-shell-word       'w'   '^[^['{\[,O}C   '^['{'[1;',\[,O}{3,5}C
+  # .edit.bind forward-shell-word     'w'  '^[^['{\[,O}B   '^['{'[1;',\[,O}{3,5}B
+  #
+  # .edit.bind backward-kill-subword    '^[^?'  '^[^?'  '^H'  '^[[27;'{3,5}';8~'
+  # .edit.bind backward-kill-shell-word '^W'    '^[^H'        '^[[27;'{6,7}';8~'
+  # bindkey '^[[3~' delete-char
+  # .edit.bind kill-subword             '^[d'   '^[(' '^[^[[3~'   '^[[3;'{3,5}\~
+  # .edit.bind kill-shell-word          '^[^D'        '^[^[[3;5~' '^[[3;'{6,7}\~
+  # .edit.bind beginning-of-line        '^A'    '^['{\[,O}H
+  # .edit.bind end-of-line              '^E'    '^['{\[,O}F
+  # .edit.bind beginning-of-buffer      '^[<'   '^['{'[1;',\[,O}5H  '^[[5~'
+  # .edit.bind end-of-buffer            '^[>'   '^['{'[1;',\[,O}5F  '^[[6~'
+  # .edit.bind backward-kill-line       '^U'    '^['{'[1;',\[,O}2H  '^['{'[1;',\[,O}2D
+  # .edit.bind kill-line                '^K'    '^['{'[1;',\[,O}2F  '^['{'[1;',\[,O}2C
+  #
+  # unfunction .edit.bind

-  unfunction .edit.bind
+  # bindkey -M emacs  '^[e' redo  '^[Y' reverse-yank-pop

-  bindkey -M emacs  '^[e' redo  '^[Y' reverse-yank-pop
+  bindkey -M vicmd 'gg' beginning-of-buffer
+  bindkey -M vicmd 'G' end-of-buffer
+  bindkey -M vicmd 'U' redo

-  bind    -M emacs      '^[-' 'pushd -1'      '^[=' 'pushd +0'
+  bind    -M viins      '^[-' 'pushd -1'      '^[=' 'pushd +0'
   bindkey -M menuselect '^[-' menu-complete   '^[=' reverse-menu-complete

   zle -N dirstack-minus .edit.dirstack
   zle -N dirstack-plus  .edit.dirstack
-  bindkey -M emacs          '^[_' dirstack-minus  '^[+' dirstack-plus
+  bindkey -M viins          '^[_' dirstack-minus  '^[+' dirstack-plus
   bindkey -M menuselect -s  '^[_' '^G^_^[_'       '^[+' '^G^_^[+'
